src/main/msc/usbd_storage_sd_spi.c

```diff
@@ -213,8 +213,9 @@ static int8_t STORAGE_Read (uint8_t lun,
 	UNUSED(lun);
 	LED1_ON;
 	for (int i = 0; i < blk_len; i++) {
-    while (sdcard_readBlock(blk_addr + i, buf + (512 * i), NULL, 0) == 0);
-    while (sdcard_poll() == 0);
+		while (sdcard_readBlock(blk_addr + i, buf + (512 * i), NULL, 0) == 0)
+			sdcard_poll();
```

Comment:
Is this extra sdcard_poll() part of the bug fixes?

Reply:
yes, should the call `sdcard_readBlock` fail (for example because the card is not done initializing), previously there was nothing to move the state machine along. the `STORAGE_Write` function in this same file has always been using the same mechanic, so i am guessing it was simply forgotten here.